.github/workflows/actionlint.yml

```diff
@@ -0,0 +1,52 @@
+name: actionlint
+
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - '.github/workflows/*.yml'
+      - '.github/workflows/*.yaml'
+  pull_request:
+    paths:
+      - '.github/workflows/*.yml'
+      - '.github/workflows/*.yaml'
+  merge_group:
+
+env:
+  HOMEBREW_DEVELOPER: 1
+
+concurrency:
+  group: "actionlint-${{ github.ref }}"
+  cancel-in-progress: ${{ github.event_name == 'pull_request' }}
+
+jobs:
+  workflow_syntax:
+    if: github.repository == 'Homebrew/brew'
+    runs-on: ubuntu-22.04
+    container:
+      image: ghcr.io/homebrew/ubuntu22.04:master
+    defaults:
+      run:
+        shell: /bin/bash -e {0}
+    steps:
+      - name: Set up Homebrew
+        id: setup-homebrew
+        uses: Homebrew/actions/setup-homebrew@master
+        with:
+          test-bot: false
+
+      - name: Set up actionlint
+        env:
+          HOMEBREW_REPOSITORY: ${{ steps.setup-homebrew.outputs.repository-path }}
+        run: |
+          brew install actionlint shellcheck
+
+          # Annotations work only relative to GITHUB_WORKSPACE
```

Comment:
> Are you referring to `add-matcher` or the process the matcher later does?

Both. Annotations work for us otherwise because we use `GitHub::Actions::Annotation`, which handles emitting the correct relative paths:

https://github.com/Homebrew/brew/blob/92d68e318d5bedf2305be35227b2157079542ce8/Library/Homebrew/utils/github/actions.rb#L72

I couldn't get this to show annotations without moving everything into `GITHUB_WORKSPACE` at Homebrew/homebrew-core#128075. I didn't bother trying here, though.

Reply:
Ok looks like this https://github.com/actions/runner/issues/765.